### ISSUE TITLE
no-void-expression does not support arrow function declaration

### ISSUE THREAD

**Issue by grimly** [palantir/tslint#3809]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.8.1
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
window.setTimeout(() => alert(50), 0);
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "no-void-expression": true
    }
}
```

#### Actual behavior

`no-void-expression` because `alert` returns void.

#### Expected behavior

No error.

**Comment by maxi7587** [comment]
I have a workaround but it does't solve the issue. When using a function that returns void:
``` javascript
.subscribe( something => someFunction(something));
```
no-voidexpression ruel alerts that `someFunction(something)` returns void. Tried to type it but it keeps launching the error. The only workaround I found is to use curly brackets:
```
.subscribe( something => { someFunction(something) });
```
works fine, but i'd like to use it without the curly brackets.

**Comment by maxi7587** [comment]
Finally have a solution!!! Add the following line in your `tslint.json` to allow arrow functions shorthand:
``` javascript
"no-void-expression": [true, "ignore-arrow-function-shorthand"],
```

**Comment by JoshuaKGoldberg** [comment]
Yup! Duplicate of #2473.
